airbyte-integrations/connectors/source-zendesk-support/source_zendesk_support/streams.py

```diff
@@ -473,7 +476,69 @@ def request_params(
         return params
 
 
-class SourceZendeskIncrementalExportStream(SourceZendeskSupportCursorPaginationStream):
+class SourceZendeskSupportCursorPaginationStream(SourceZendeskSupportFullRefreshStream):
+    """
+    Only used by streams modified by rudderstack
+    Implements cursor pagination as most streams support cursor pagination.
+    Where it is not supported such streams will override some of this functionality to
+    use offset pagination.
+    """
+
+    cursor_field = "updated_at"
+    next_page_field = "next_page"
+    prev_start_time = None
+
+    @property
+    def state_checkpoint_interval(self) -> Optional[int]:
+        """
+        Will allow the connector send state messages more frequent and not only at the end of the sync.
+        """
+        return 1000
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
```

Comment:
Do we need this method? SourceZendeskSupportStream has the same method with the same implementation,

Reply:
The pagination classes have parent `BaseSourceZendeskSupportStream` which does not have this functionality. The naming of classes is misleading here..